dyno/microcontroller/Session.h

```diff
@@ -6,32 +6,43 @@
 #define SESH_H
 #include "constants.h"
 
-class Session {
-   public:
-       //interface
-       Session(const int POT_PIN, const int* DRV_PIN, bool paused, bool drive_w_pot=false);
-       void init_temp_sense(const int *TEMP_PINS);
-       void check_for_command();
-       bool check_paused();
-       void read_pot();
-       void read_temp();
-       void drive_magnet();
-       void print_report();
-   private:
-       //private functions
-       void process_command();
-       int ramp_signal(int sig_in);
-      
-       //data members
-       int m_pot_pin;       
-       int m_temp_pins[NUM_TEMP_SENSE];
-       float m_temps[NUM_TEMP_SENSE];
-       int m_drv_pins[8];
-       int m_pwms[NUM_MAGNETS];
-       char m_com_buff[256];
-       int m_com_buff_count;
-       bool m_paused_flag;
-       bool m_drive_w_pot;
+struct SerialStats
+{
+    int m_chars_received = 0;
+    int m_commands_received = 0;
+    int m_commands_parsed = 0;
+    int m_commands_not_parsed = 0;
+};
```

Comment:
The diff here isn't too helpful, but the only changes made were adding this struct declaration and including it as a data member in the `Session` class below.

Reply:
Love this